Firestore/core/src/firebase/firestore/util/error_apple.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_ERROR_APPLE_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_ERROR_APPLE_H_
+
+// Everything in this header exists for compatibility with Objective-C.
+#if __OBJC__
+
+#import <Foundation/Foundation.h>
+
+#include "Firestore/Source/Public/FIRFirestoreErrors.h"  // for FIRFirestoreErrorDomain
+#include "Firestore/core/include/firebase/firestore/firestore_errors.h"
+#include "Firestore/core/src/firebase/firestore/util/string_apple.h"
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+// Translates a set of error_code and error_msg to an NSError.
+inline NSError* WrapNSError(const int64_t error_code,
+                            const absl::string_view error_msg) {
+  if (error_code == FirestoreErrorCode::Ok) {
+    return nil;
+  }
+  return [NSError
+      errorWithDomain:FIRFirestoreErrorDomain
+                 code:error_code
+             userInfo:@{
+               NSLocalizedDescriptionKey : WrapNSStringNoCopy(error_msg)
```

Comment:
This should be a `WrapNSString` (i.e. a copy). The point of wrapping a NSError up is frequently to return it to a caller or otherwise allow it to indicate status in a way that may escape the the lifetime of the calling frame. `WrapNSStringNoCopy` is only reasonable when the lifetime of the string is bounded within the lifetime of the caller.